xterm-color.el

```diff
@@ -741,6 +741,21 @@ prefix arg causes overlays to be used."
     (goto-char (point-min))
     (when read-only-p (read-only-mode 1))))
 
+;;;###autoload
+(defun xterm-color-apply-on-region (beg end &optional use-overlays)
+  "Apply `xterm-color-filter' to current region.
+
+When called non-interactively, apply to region bounded by buffer
+positions BEG and END. See `xterm-color-colorize-buffer' for the
+meaning of USE-OVERLAYS."
+  (interactive "r")
+  (let ((pos (point)))
+    (save-restriction
+      (narrow-to-region beg end)
+      (let ((buffer-read-only nil))
+        (xterm-color-colorize-buffer use-overlays)))
```

Comment:
I don't plan on merging this, but as a courtesy:

To improve safety, you need to wipe all xterm-color-specific buffer-local state before and after you call this function.
You also need to check after every call if it has produced buffer-local state at which point you need to let the user know that things are probably NOT OK.